provider.go

```diff
@@ -373,6 +373,9 @@ type Provider struct {
 
 	// Components Resources
 	Construct func(context.Context, ConstructRequest) (ConstructResponse, error)
+
+	Name    string
+	Version string
```

Comment:
This change needs a bit more thought in my opinion. We can think about that in a separate PR by changing the signature of `(Provider).Run` to `func(ctx context.Context, name, version string) error` for now.

Reply:
That's fine I guess, I just really want to avoid a final API that looks like this:

```go
p, _ := builder.WithName(name).WithVersion(version).Build()
p.Run(ctx, name, version)
```

An alternative would be to drop `WithName` and `WithVersion` but I probably prefer to keep them.

AFAICT name is only used in pu/pu for tracing, and go-provider is adding them to the context's RuntimeInfo.